File: backend_go/models/bot.go
package models

import "time"

type Bot struct {
	ID                 uint      `gorm:"primaryKey"`
	OwnerID            *uint     `gorm:"index"` // Nullable, for referral bots
	Token              string    `gorm:"unique"`
	Username           string    `gorm:"unique"`
	Type               string    `gorm:"index"` // 'main' or 'referral'
	IsActive           bool      `gorm:"default:true"`
	IsPrimary          bool      `gorm:"default:false"` // For referral bots
	ReferralPercentage float64   `gorm:"default:0"`     // For referral bots
	CreatedAt          time.Time `gorm:"autoCreateTime"`
}

type BotResponse struct {
	ID                 uint    `json:"id"`
	Token              string  `json:"token"`
	Username           string  `json:"username"`
	Type               string  `json:"type"`
	IsPrimary          bool    `json:"is_primary"`
	IsActive           bool    `json:"is_active"`
	OwnerID            *uint   `json:"owner_id"`
	OwnerTelegramID    int64   `json:"owner_telegram_id,omitempty"`
	Turnover           float64 `json:"turnover"`
	Accruals           float64 `json:"accruals"`
	ReferralPercentage float64 `json:"referral_percentage"`
}

File: backend_rust/src/presentation/bot/dtos/bot.rs
use shared_dtos::bot::BotBotResponse;

use crate::models::bot::BotRow;
use rust_decimal::prelude::ToPrimitive;

impl From<BotRow> for BotBotResponse {
    fn from(r: BotRow) -> Self {
        BotBotResponse {
            id: r.id,
            token: r.token,
            username: r.username,
            is_active: r.is_active,
            is_primary: r.is_primary,
            referral_percentage: r.referral_percentage.to_f64().unwrap_or_default(),
            owner_id: r.owner_id,
        }
    }
}
